libs/ledger-live-common/src/families/stellar/api/horizon.ts

```diff
@@ -107,24 +121,25 @@ export const fetchAccount = async (
   spendableBalance: BigNumber;
   assets: BalanceAsset[];
 }> => {
-  let account: typeof AccountRecord = {};
-  let balance: Record<string, any> = {};
+  let account: ServerApi.AccountRecord = {} as ServerApi.AccountRecord;
   let assets: BalanceAsset[] = [];
+  let balance = "0";
 
   try {
     account = await server.accounts().accountId(addr).call();
-    balance = account.balances?.find(balance => {
-      return balance.asset_type === "native";
-    });
+    balance =
+      account.balances.find(balance => {
+        return balance.asset_type === "native";
+      })?.balance || "0";
     // Getting all non-native (XLM) assets on the account
-    assets = account.balances?.filter(balance => {
+    assets = account.balances.filter(balance => {
       return balance.asset_type !== "native";
-    });
+    }) as BalanceAsset[];
```

Comment:
Yeah, I think it's a legacy thing but the type of `account.balances` is `Horizon.BalanceLine` but the function expects assets to be of type `BalanceAsset` it demanded too much change so I decided to keep the `as` for now

Reply:
I think the former developers weren't aware that the type of `assets` was available so they created custom types
It would be a lot more changes just to remove `BalanceAsset` type everywhere